source/slang/slang-compiler-tu.cpp

```diff
@@ -59,27 +61,76 @@ namespace Slang
         CodeGenContext::Shared sharedCodeGenContext(&tp, entryPointIndices, &sink, nullptr);
         CodeGenContext codeGenContext(&sharedCodeGenContext);
 
-        ComPtr<IArtifact> outArtifact;
-        SlangResult res = codeGenContext.emitTranslationUnit(outArtifact);
-
-        sink.getBlobIfNeeded(outDiagnostics);
+        // Mark all public functions as exported, ensure there's at least one
+        bool hasAtLeastOneFunction = false;
 
-        if (res != SLANG_OK)
+        for (auto inst : module->getGlobalInsts())
         {
-            return res;
+            if (inst->getOp() == kIROp_Func)
+            {
+                bool hasResourceType = false;
+
+                // DXIL does not permit HLSLStructureBufferType in exported functions
```

Comment:
This won't work if there is a struct type containing the resource. The right way is to mark all functions as "attempt to export" now, and make them real "export" after all the legalization passes if the function body meets requirements.

Reply:
I'm considering a new decoration like "TransientExport" that I can treat like "HLSLExport" in "_isHLSLExported" but would be easy to remove after.

Reply:
When marking functions as "real export", please confirm if that means adding kIROp_HLSLExportDecoration.  Also, is the right time to do that during linkAndOptimizeIR() after linkIR()?  That's where I'm looking right now, but my intuition is that I need to inspect IR before lowerGenerics() is called, however that is before much legalization.